### PR TITLE
[DEV-1431] Integration of API client

### PR DESCRIPTION
**Pull request by bwolkerstorfer**
Ticket: DEV-1431

Change Log:
* fixed issues with models, requests and responses
* added services for API resources (DI-oriented)
* added tests for servics (integrated with mock server)

### CODE REVIEW

File: src/eRecruiter.Api/Services/ApiBaseService.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    {
+        #region Properties
+
+        protected ApiClient ApiClient { get; }
+
+        #endregion
+
+        #region Constructor
+
+        private readonly string _resource;
```
Comment (kopfrechner) [inline src/eRecruiter.Api/Services/ApiBaseService.cs]: How about 'protected'. This string could be reused in URI's of derivied serivces too.

Comment (bwolkerstorfer) [inline src/eRecruiter.Api/Services/ApiBaseService.cs]: switched to use the resource identifier with commit dab2e4f

File: src/eRecruiter.Api/Services/Group/GroupService.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        #region Asynchronous Manipulation
+
+        /// <summary>
+        /// Add recruiting agencies to the given group.
+        /// </summary>
+        /// <param name="groupId">Group ID</param>
+        /// <param name="recruitingAgencyIds">Recruiting Agency IDs</param>
+        /// <returns>recruiting agency IDs currently in the group</returns>
+        public async Task<IEnumerable<int>> AddRecruitingAgenciesToGroupAsync(int groupId, IEnumerable<int> recruitingAgencyIds)
```
Comment (kopfrechner) [inline src/eRecruiter.Api/Services/Group/GroupService.cs]: There was a wrong naming in the API specification -> This should be called
SynchronizeRecruitingAgenciesWithGroupAsync

File: src/eRecruiter.Api/Services/Group/GroupService.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            return await ApiClient.SendRequestAsync<IEnumerable<int>, IEnumerable<int>>(HttpMethod.Put, $"/{_resource}/{groupId}/recruitingAgencies", recruitingAgencyIds);
+        }
+
+        /// <summary>
+        /// Add recruiters to the given group.
+        /// </summary>
+        /// <param name="groupId">Group ID</param>
+        /// <param name="recruiterIds">Recruiter IDs</param>
+        /// <returns>recruiters currently in the group</returns>
+        public async Task<IEnumerable<int>> AddRecruitersToGroupAsync(int groupId, IEnumerable<int> recruiterIds)
```
Comment (kopfrechner) [inline src/eRecruiter.Api/Services/Group/GroupService.cs]: There was a wrong naming in the API specification -> This should be called
SynchronizeRecruitersWithGroupAsync

File: src/eRecruiter.Api/Services/Group/GroupService.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            return AddRecruitingAgenciesToGroupAsync(groupId, recruitingAgencyIds).Result;
+        }
+
+        /// <summary>
+        /// Add recruiters to the given group.
+        /// </summary>
+        /// <param name="groupId">Group ID</param>
+        /// <param name="recruiterIds">Recruiter IDs</param>
+        /// <returns>recruiters currently in the group</returns>
+        public IEnumerable<int> AddRecruitersToGroup(int groupId, IEnumerable<int> recruiterIds)
```
Comment (kopfrechner) [inline src/eRecruiter.Api/Services/Group/GroupService.cs]: There was a wrong naming in the API specification -> This should be called
SynchronizeRecruitingAgenciesWithGroup

File: src/eRecruiter.Api/Services/Group/GroupService.cs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+        #region Manipulation
+
+        /// <summary>
+        /// Add recruiting agencies to the given group.
+        /// </summary>
+        /// <param name="groupId">Group ID</param>
+        /// <param name="recruitingAgencyIds">Recruiting Agency IDs</param>
+        /// <returns>recruiting agency IDs currently in the group</returns>
+        public IEnumerable<int> AddRecruitingAgenciesToGroup(int groupId, IEnumerable<int> recruitingAgencyIds)
```
Comment (kopfrechner) [inline src/eRecruiter.Api/Services/Group/GroupService.cs]: There was a wrong naming in the API specification -> This should be called
SynchronizeRecruitingAgenciesWithGroup

File: test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        {
+            Delete();
+        }
+
+        #endregion
+
+        #region Manipulation
+
+        [Fact]
+        public void AddRecruitingAgenciesToGroupTest()
```
Comment (kopfrechner) [inline test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs]: Rename to SynchronizeRecruitingAgenciesWithGroup

File: test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        {
+            var result = Service.AddRecruitingAgenciesToGroup(1, new[] {1, 2, 3}).ToList();
```
Comment (kopfrechner) [inline test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs]: Use Synchronize Method

File: test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+            // assertions
+            Assert.NotNull(result);
+            Assert.NotEmpty(result);
+            Assert.Equal(3, result.Count);
+        }
+
+        [Fact]
+        public void AddRecruitersToGroupTest()
```
Comment (kopfrechner) [inline test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs]: Rename to SynchronizeRecruitersWithGroup

File: test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        {
+            var result = Service.AddRecruitersToGroup(1, new[] {1, 2, 3}).ToList();
```
Comment (kopfrechner) [inline test/eRecruiter.Api.Test/Integration/GroupServiceTests.cs]: Use Synchronize Method

**Comment by bwolkerstorfer** [comment]
@kopfrechner The changes have been made and PR ready for another round.

**Pull request merged**